modules/GKB/EnsEMBLMartUtils.pm

```diff
@@ -48,44 +50,46 @@ our %EXPORT_TAGS = (all => [@EXPORT_OK],
                    query => [qw/get_query get_query_runner/]);
 
 sub get_species_results {
-    my $species = shift;
-    return get_species_results_with_attribute_info($species)->{'results'};
+    my $logger = get_logger(__PACKAGE__);
+
+    my $species_abbreviation = shift // $logger->logconfess("Need species abbreviation");
+    return get_species_results_with_attribute_info($species_abbreviation)->{'results'};
 }
 
 sub get_species_results_with_attribute_info {
-    my $species = shift;
-    
     my $logger = get_logger(__PACKAGE__);
-    
+
+    my $species_abbreviation = shift // $logger->logconfess("Need species abbreviation");
+
     my $species_results_with_attribute_info;
     my $results_complete;
     my $query_attempts = 0;
     until (($species_results_with_attribute_info && $results_complete) || $query_attempts == 3) {
         $query_attempts += 1;
-        
+
         my $five_minutes = 5 * 60;
         timeout $five_minutes => sub {
-            $species_results_with_attribute_info = query_for_species_results($species);
+            $species_results_with_attribute_info = query_for_species_results($species_abbreviation);
         };
 
         $results_complete = $species_results_with_attribute_info->{'results'} =~ /\[success\]$/;
-                            
-        $logger->info("Query attempt $query_attempts for species $species");
-        $results_complete ? 
-            $logger->info("Results obtained successfully") : 
+
+        $logger->info("Query attempt $query_attempts for species $species_abbreviation");
+        $results_complete ?
+            $logger->info("Results obtained successfully") :
             $logger->warn("Problem obtaining results - got " . $species_results_with_attribute_info->{'results'});
     }
-    
+
     return $species_results_with_attribute_info;
 }
 
 sub query_for_species_results {
-    my $species = shift;
-    my $uniprot_attribute_info = shift // {};
-
     my $logger = get_logger(__PACKAGE__);
 
-    my $mart_info = get_mart_info_for_species($species);
+    my $species_abbreviation = shift // $logger->logconfess("Need species abbreviation");

```

Comment:
(see comment I made earlier about aborting a function when input is null, instead of logging and continuing like everything is OK)

Reply:
Apparently, `logconfess` _will_ break execution, in addition to logging a message.